examples/mne_util.py

```diff
@@ -0,0 +1,294 @@
+                roiidx.extend(np.full(lverts.size, li, dtype=np.int64))
+
+                # add 1 b/c 0 corresponds to unassigned variance
+                Q_J[lverts] = li + 1
+
+        N = len(labels)
+        M = n_verts
+
+        # construct sparse L matrix
+        data = np.concatenate(data)
+        vertidx = np.array(vertidx, int)
+        roiidx = np.array(roiidx, int)
+        assert data.shape == vertidx.shape == roiidx.shape
+        L = csc_matrix((data, (vertidx, roiidx)), shape=(M, N))
+
+        self.fwd = fwd
+        self.L = L
+        self.Q_J = Q_J
+        self.offsets = offsets
+        self.n_lhverts = n_lhverts
+        self.n_rhverts = n_rhverts
+        self.labels = labels
+
+        return
+
+    @property
+    def G(self):
+        return self.fwd['sol']['data']
+
+    @property
+    def L(self):
+        return self._L
+
+    @L.setter
+    def L(self, val):
+        self._L = val
+
+    @property
+    def Q_J(self):
+        return self._Q_J
+
+    @Q_J.setter
+    def Q_J(self, val):
+        self._Q_J = val
+
+    @property
+    def GL(self):
+        from util import Carray
+        return Carray(csr_matrix.dot(self.L.T, self.G.T).T)
+
+    def get_label_vinds(self, label):
+        li = self.labels.index(label)
+        if isinstance(label, mne.Label):
+            label_vert_idx = self.L[:, li].nonzero()[0]
+            label_vert_idx -= self.offsets[label.hemi]
+            return label_vert_idx
+        elif isinstance(label, mne.BiHemiLabel):
+            # these labels store both hemispheres so subtract the rh offset
+            # from that part of the vertex array
+            lh_label_vert_idx = self.L[:self.n_lhverts, li].nonzero()[0]
+            rh_label_vert_idx = self.L[self.n_lhverts:, li].nonzero()[0]
+            rh_label_vert_idx[self.n_lhverts:] -= self.offsets['rh']
+            return [lh_label_vert_idx, rh_label_vert_idx]
+
+    def get_label_verts(self, label, src):
+        # if you're thinking of using this to plot, why not just use
+        # brain.add_label from pysurfer?
+        if isinstance(label, mne.Label):
+            hi = 0 if label.hemi == 'lh' else 1
+            label_vert_idx = self.get_label_vinds(label)
+            varray = src[hi]['vertno'][label_vert_idx]
+        elif isinstance(label, mne.BiHemiLabel):
+            lh_label_vert_idx, rh_label_vert_idx = self.get_label_vinds(label)
+            varray = [src[0]['vertno'][lh_label_vert_idx],
+                      src[1]['vertno'][rh_label_vert_idx]]
+        return varray
+
+    def get_hemi_idx(self, label):
+        if isinstance(label, mne.Label):
+            return 0 if label.hemi == 'lh' else 1
+        elif isinstance(label, mne.BiHemiLabel):
+            hemis = [None] * 2
+            for i, lab in enumerate([label.lh, label.rh]):
+                hemis[i] = 0 if lab.hemi == 'lh' else 1
+            return hemis
+
+
+def morph_labels(labels, subject_to, subjects_dir=None):
```

Comment:
Let's remove this for now, does not seem relevant/needed

Reply:
Is this just in reference to `def morph_labels` or the entirety of lines 91-151? I'm still getting used to reading Github.